src/bundle-stats.js

```diff
@@ -100,11 +101,11 @@ BundleStatsCollector.prototype.LoadStatsFromDisk = function (outputdirectory) {
         return ret;
     }
 
-    _this.HashCollection = loadFromDisk(_this.FileSystem, outputdirectory, getFileName(this,HASH_FILE_NAME));
-    _this.DebugCollection = loadFromDisk(_this.FileSystem, outputdirectory, getFileName(this,DEBUG_FILE_NAME));
-    _this.LocalizedStrings = loadFromDisk(_this.FileSystem, outputdirectory, getFileName(this,LOCALIZATION_FILE_NAME));
-    _this.AbConfigs = loadFromDisk(_this.FileSystem, outputdirectory, getFileName(this,AB_FILE_NAME));
-    _this.LessImports = loadFromDisk(_this.FileSystem, outputdirectory, getFileName(this,LESS_IMPORTS_FILE));
+    _this.HashCollection = collection.createHash(loadFromDisk(_this.FileSystem, outputdirectory, getFileName(this,HASH_FILE_NAME)));

```

Comment:
Refactored this file slightly to use a custom JS collection type rather than simple arrays so we can add some safety checks when items are added.